### ISSUE TITLE
Global package functions and design issues.

### ISSUE THREAD

**Issue by mephux** [albrow/zoom#16]
So, in my usecase I need to have two connection pools, both connected to different databases. The current design of the package makes this impossible. IMHO, Init should return a *zoom.Connection and have everything else live under it.

```go

// pool 1
connection, err := zoom.Init(...)
defer connection.Close()

// pool 2
connection2, err := zoom.Init(...)
defer connection2.Close()

// ...
connection.Models()
connection.Query.blah

connection.Register(OtherModel)
connection2.Register(Model)
```

I forked and changed all the code but the tests were designed in a way that made it impossible so I just removed my work and decided to go back another redis client lib.

I would also suggest using TestMain() vs 

```go
	testingSetUp()
	defer testingTearDown()
```

Anyway, I really like the concepts of this project and I hope these types of changes will be worked on in the future. If i get more time I may try to continue the test fixes but I fear it would be a total rewrite to make it work the way I want it to work.

Best of luck.

**Comment by albrow** [comment]
@mephux thanks for sharing your use case. I had never thought about someone needing two different connection pools and in never came up in my own projects, but that does make sense. This change actually seems pretty straightforward and I'll plan on implementing it before version 1.0.

Keep an eye out for a PR soon. I would appreciate you reviewing it to make sure it meets your needs. I'll look into using TestMain as well.

**Comment by mephux** [comment]
@albrow awesome, will do. Keep up the great work.

**Comment by albrow** [comment]
@mephux please check out https://github.com/albrow/zoom/pull/17. Still need to update doc comments and README but it looks like it works and all the tests pass.

**Comment by albrow** [comment]
@mephux it's been 7 days since I fixed this and created PR #17 and I haven't heard from you. I'm going to go ahead and assume that the new implementation works for you and merge it into master for release 0.10.0 soon.

**Comment by mephux** [comment]
@albrow so sorry for the delayed reply. It turned out to be a very busy end of month for me. I thought I had replied before but guess i didn't in error. Everything looks great to me tho. Awesome work.
